languages/ruby/exercises/concept/basics/.docs/instructions.md

```diff
@@ -0,0 +1,42 @@
+In this exercise you're going to write some code to help you cook a brilliant lasagna from your favorite cooking book.
+
+You have four tasks, all related to the time spent cooking the lasagna.
+
+### 1. Define the expected oven time in minutes
+
+Define the `Lasagna::EXPECTED_MINUTES_IN_OVEN` constant that returns how many minutes the lasagna should be in the oven. According to the cooking book, the expected oven time in minutes is 40:
+
+```ruby
+Lasagna::EXPECTED_MINUTES_IN_OVEN
+# => 40
+```
+
+### 2. Calculate the remaining oven time in minutes
+
+Define the `Lasagna#remaining_minutes_in_oven` method that takes the actual minutes the lasagna has been in the oven as a parameter and returns how many minutes the lasagna still has to remain in the oven, based on the expected oven time in minutes from the previous task.
+
+```ruby
+lasagna = Lasagna.new
+lasagna.remaining_minutes_in_oven(30)
+# => 10
+```
+
+### 3. Calculate the preparation time in minutes
+
+Define the `Lasagna#preparation_time_in_minutes` method that takes the number of layers you added to the lasagna as a parameter and returns how many minutes you spent preparing the lasagna, assuming each layer takes you 2 minutes to prepare.
+
+```ruby
+lasagna = Lasagna.new
+lasagna.preparation_time_in_minutes(2)
+# => 4
+```
+
+### 4. Calculate the total working time in minutes
+
+Define the `Lasagna#total_time_in_minutes` method that takes two named parameters: the `number_of_layers` parameter is the number of layers you added to the lasagna, and the `actual_minutes_in_oven` parameter is the number of minutes the lasagna has been in the oven. The function should return how many minutes in total you've worked on cooking the lasagna, which is the sum of the preparation time in minutes, and the time in minutes the lasagna has spent in the oven at the moment.
```

Comment:
Parameters may suggest something a little differently for those that are coming from a CS background and other 
programming languages.  Should we refer to the argument list for methods to generally be arguments?  The documentation, I believe, generally refers to them as this as well.

Reply:
Isn't an argument a value to be passed as a parameter? But we should indeed conform to the documentation (I don't know how Ruby does this by the way).

Reply:
I've just checked and what I called called `named parameters` are actually called `keyword arguments`, so let's update everywhere to use that.

Reply:
👍

Reply:
Yes the word parameter suggests a limitation or set of limitations, we do things within parameters of the problem, for example.